[docs] Configure and integrate Vale in docs (#24146)
expo/expo: docs/README.md

yarn lint
```

### Prose linter

We use [Vale](https://vale.sh/) to lint our docs for style and grammar based on [Expo's writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md).

There are two ways you can use it:

> #### Use Vale in VSCode (Recommended)

Author: ```suggestion
#### Use Vale in VS Code (Recommended)
```

> #### Use Vale in VSCode (**Recommended**)

Author: ```suggestion
#### Use Vale in VSCode (Recommended)
```

- [Install Vale on your system](https://vale.sh/docs/vale-cli/installation/)

> - [Install Vale's VSCode extension](https://marketplace.visualstudio.com/items?itemName=errata-ai.vale-server)

Author: ```suggestion
- [Install Vale's VS Code extension](https://marketplace.visualstudio.com/items?itemName=errata-ai.vale-server)
```

> 

Author: ```suggestion

Open the doc file (`*.mdx`) that you are working on and you'll may see suggested lines (yellow squiggly) in VS Code editor.
```

#### Run the `lint-prose` script

> Run the `yarn run lint-prose` script from **package.json**. This will run Vale on all markdown files in the **pages** directory.

Author: ```suggestion
In a terminal window, run the `yarn run lint-prose` script from **package.json**. This will run Vale for all markdown files in the **pages** directory.
```
